base/reflection.jl

```diff
@@ -212,13 +212,18 @@ end
 
 Get the name of field `i` of a `DataType`.
 
+The return type is `Int` (the index of the field) for `x <: Tuple`, otherwise the return type is `Symbol`.
```

Comment:
As for my other remark, I would suggest having the "common case" first.

Anyway, good idea to document this and to add the examples, thanks!